src/kilo.c

```diff
@@ -875,10 +875,15 @@ void editorMoveCursor(enum DIRECTION dir) {
 }
 
 void editorMoveCursorToRowEnd() {
-    int size = E.row[E.rowoff+E.cy].size;
-    while (size != 0) {
+    int filerow = E.rowoff+E.cy;
+    int filecol = E.coloff+E.cx;
+    erow *row = (filerow >= E.numrows) ? NULL : &E.row[filerow];
+    int rowlen = row ? row->size : 0;
+    int size = rowlen - E.cx-E.coloff;
```

Comment:
if row == NULL, this will be equal to (0 - E.cx) - E.coloff, so the loop below will iterate a lot :)